src/view/provider/selected-devices/__tests__/useSelectedDevicesContext.test.ts

```diff
@@ -0,0 +1,73 @@
+import { act, renderHook } from "@testing-library/react";
+import { CountryName } from "../../../../data/constants/CountryEmissions";
+import { areMapsDeepEqual } from "../../../../utils/helpers/areMapsDeepEqual";
+import { useSelectedDevicesContext } from "../useSelectedDevicesContext";
+
+describe("useSelectedDevicesContext", () => {
+    const mockCountries: Map<CountryName, number> = new Map([]);
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it("addSelectedDevice should update selectedDevices", async () => {
+        const { result } = renderHook(useSelectedDevicesContext);
+        const mockDevice1 = "iPhone 12";
+        const mockDevice2 = "Google Pixel 6";
+
+        mockCountries.set(mockDevice1, 0);
+        await act(async () => {
+            await result.current.addSelectedDevice(mockDevice1);
+        });
+        expect(result.current.selectedDevices).toStrictEqual(mockCountries);
+
+        mockCountries.set(mockDevice2, 0);
+        await act(async () => {
+            await result.current.addSelectedDevice(mockDevice2);
+        });
+        expect(result.current.selectedDevices).toStrictEqual(mockCountries);
+    });
+
+    it("removeSelectedCountries should update selectedDevices", async () => {
+        const { result } = renderHook(useSelectedDevicesContext);
+
+        const mockDevice1 = "iPhone 12";
+        const mockDevice2 = "Google Pixel 6";
+
+        mockCountries.set(mockDevice1, 0);
+        mockCountries.set(mockDevice2, 0);
+
+        await act(async () => {
+            await result.current.addSelectedDevice(mockDevice1);
+        });
+
+        await act(async () => {
+            await result.current.addSelectedDevice(mockDevice2);
+        });
+
+        expect(
+            areMapsDeepEqual(result.current.selectedDevices, mockCountries)
+        ).toBe(true);
+
+        mockCountries.delete(mockDevice2);
```

Comment:
not sure why this is needed

Reply:
should be other test

Reply:
updated name of test for readability - I think the expect statement after adding should stay here, so the test flow is:
1. add countries
2. expect countries to be added
3. delete country
4. expect country to be deleted